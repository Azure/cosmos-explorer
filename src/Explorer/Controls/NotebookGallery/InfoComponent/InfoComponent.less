@import "../../../../../less/Common/Constants.less";
.infoPanel, .infoPanelMain {
    display: flex;
    align-items: center;
}

.infoPanel {
    padding-left: 5px;
    padding-right: 5px;
}

.infoLabel, .infoLabelMain {
    padding-left: 5px  
}

.infoLabel {
    font-weight: 400
}

.infoIconMain {
    color: @AccentMedium
}

.infoIconMain:hover {
    color: @BaseMedium
}