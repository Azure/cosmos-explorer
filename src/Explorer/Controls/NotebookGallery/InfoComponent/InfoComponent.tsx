import * as React from "react";
import { Icon, Label, Stack, HoverCard, HoverCardType, Link } from "office-ui-fabric-react";
import { CodeOfConductEndpoints } from "../../../../Common/Constants";
import "./InfoComponent.less";

export interface InfoComponentProps {
  onReportAbuseClick?: () => void;
}

export class InfoComponent extends React.Component<InfoComponentProps> {
  private getInfoPanel = (iconName: string, labelText: string, url?: string, onClick?: () => void): JSX.Element => {
    return (
      <Link href={url} target={url && "_blank"} onClick={onClick}>
        <div className="infoPanel">
          <Icon iconName={iconName} styles={{ root: { verticalAlign: "middle" } }} />
          <Label className="infoLabel">{labelText}</Label>
        </div>
      </Link>
    );
  };

  private onHover = (): JSX.Element => {
    return (
      <Stack tokens={{ childrenGap: 5, padding: 5 }}>
        <Stack.Item>{this.getInfoPanel("Script", "Code of Conduct", CodeOfConductEndpoints.codeOfConduct)}</Stack.Item>
        <Stack.Item>
          {this.getInfoPanel("RedEye", "Privacy Statement", CodeOfConductEndpoints.privacyStatement)}
        </Stack.Item>
        <Stack.Item>
          {this.getInfoPanel("KnowledgeArticle", "Microsoft Terms of Use", CodeOfConductEndpoints.termsOfUse)}
        </Stack.Item>
        {this.props.onReportAbuseClick && (
          <Stack.Item>
            {this.getInfoPanel("ReportHacked", "Report Abuse", undefined, () => this.props.onReportAbuseClick())}
          </Stack.Item>
        )}
      </Stack>
    );
  };

  public render(): JSX.Element {
    return (
      <HoverCard plainCardProps={{ onRenderPlainCard: this.onHover }} instantOpenOnClick type={HoverCardType.plain}>
        <div className="infoPanelMain">
          <Icon className="infoIconMain" iconName="Help" styles={{ root: { verticalAlign: "middle" } }} />
          <Label className="infoLabelMain">Help</Label>
        </div>
      </HoverCard>
    );
  }
}
