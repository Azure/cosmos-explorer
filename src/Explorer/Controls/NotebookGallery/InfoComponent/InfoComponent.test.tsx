import { shallow } from "enzyme";
import React from "react";
import { InfoComponent } from "./InfoComponent";

describe("InfoComponent", () => {
  it("renders", () => {
    const wrapper = shallow(<InfoComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
