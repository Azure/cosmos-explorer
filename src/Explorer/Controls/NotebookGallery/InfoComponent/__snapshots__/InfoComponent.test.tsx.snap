// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoComponent renders 1`] = `
<StyledHoverCardBase
  instantOpenOnClick={true}
  plainCardProps={
    {
      "onRenderPlainCard": [Function],
    }
  }
  type="PlainCard"
>
  <div
    className="infoPanelMain"
    tabIndex={0}
  >
    <Icon
      className="infoIconMain"
      iconName="Help"
      styles={
        {
          "root": {
            "verticalAlign": "middle",
          },
        }
      }
    />
    <StyledLabelBase
      className="infoLabelMain"
    >
      Help
    </StyledLabelBase>
  </div>
</StyledHoverCardBase>
`;
